mito-ai/src/Extensions/SettingsManager/profiler/ProfilerPage.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Saga Inc.
+ * Distributed under the terms of the GNU Affero General Public License v3.0 License.
+ */
+
+import React from 'react';
+import { IContextManager } from '../../ContextManager/ContextManagerPlugin';
+import { ContextManagerSection } from './ContextManagerSection';
+import { CapturedRequestsSection } from './CapturedRequestsSection';
+import '../../../../style/ProfilerPage.css';
+
+interface ProfilerPageProps {
+    contextManager: IContextManager;
+}
+
+// Simple event system for capturing completion requests
+export const captureCompletionRequest = (request: any): void => {
+    const event = new CustomEvent('mito-ai-capture-request', {
```

Comment:
Let's make `'mito-ai-capture-request'` a constant somewhere so we're sure to always keep it in sync